### ISSUE TITLE
Slow `read_parquet` with many files.

### ISSUE THREAD

**Issue by TomAugspurger** [geopandas/dask-geopandas#194]
I'm looking into the performance of `dask_geopandas.read_parquet` when we have a Parquet dataset with many files. Dropping some notes here.

## Context

I'm working on Azure Blob Storage, but this should apply to S3 and GCS as well.

The structure is

```
/path/to/root.parquet
    /region=A
      - part.0.parquet
      - part.1.parquet
    /region=B
      - part.0.parquet
      - part.1.parquet
      - ...
...
```

There are 206 "regions" and 3,679 parts in total.

At the moment, there's no `_metadata` or `_global_metadata` file present. I could probably add one if that's helpful.

## Minimal Example

```python
import dask_geopandas
import requests
import adlfs

token = requests.get("https://planetarycomputer-staging.microsoft.com/api/sas/v1/token/bingmlbuildings/footprints").json()["token"]

storage_options=dict(account_name="bingmlbuildings", credential=token)

%time df = dask_geopandas.read_parquet("az://footprints/geo/2022-05-25/ml-buildings.parquet", storage_options=storage_options)
````

From the same Azure region as the data, that takes about 3 minutes to complete.

## Reasons

I think most of the time is being spent in https://github.com/geopandas/dask-geopandas/blob/16561258318d2d7a2ae24c06741019050b3645f2/dask_geopandas/io/parquet.py#L26-L34 (need to confirm this)

What are the consequences of not providing that metadata? That we end up with incorrect / not `spatial_partitions` I suspect? Depending on the use case, unset spatial divisions might be OK. Would it be acceptable to disable those boundary reads with a flag in `read_parquet`?

## Relevant links:

- https://github.com/dask/dask/issues/8058

**Comment by jorisvandenbossche** [comment]
Reading this metadata is indeed only for populating the `spatial_partitions`

Adding an option to not load the spatial partitioning information upon construction of the dask objects sounds as a good idea.

On the actual implementation, I am wondering if this can actually be improved. In the parent ArrowDatasetEngine class from dask, we actually do have a pyarrow dataset object and fragments, but after `read_metadata`  dask typically extracts the file path for each fragment. If we would have more control over this, we could access the schema directly from the fragment. Although I am not sure how much benefit this will give, since accessing the schema can still invoke IO at that point to read the physical schema from each file (very similar as how gathering the statistics in dask's read_parquet gives a lot of IO when constructing the object, I _think_ this is no longer done by default?). 

More fundamentally, I am wondering to what extent we need this spatial partitioning information _up front_ (but that's a more technical dask internals question). Currently, the `spatial_partitions` information is part of the dask GeoDataFrame object, so we can use this information when constructing the task graph for eg a spatial join (to have a simplified task graph, removing the parts that will never have interaction in a spatial join). 
Would it be possible to make this `spatial_partitions` lazy? That reading this in / computing it becomes part of the task graph, and that the graph is simplified "on the fly"? 
(that's probably a discussion for a separate issue)

**Comment by TomAugspurger** [comment]
> since accessing the schema can still invoke IO at that point to read the physical schema from each file (very similar as how gathering the statistics in dask's read_parquet gives a lot of IO when constructing the object, I think this is no longer done by default?).

My understanding might be out of date, but I think there's a plan to start using Dask to do IO in parallel from the filesystem to gather these statistics up front when a `_common_metadata` file isn't present. So `read_parquet(...)` would actually eagerly execute some tasks in parallel.

IMO, spatial partitions should be similar to `divisions` (which I think is the current behavior). Not necessarily required for anything, but a nice optimization that's ideally extends to an on-disk format..
